// ngx-quicklink/esm2020/lib/quicklink.module.mjs
import { NgModule } from '@angular/core';
import { LinkDirective } from './link.directive';
import { ObservableLinkHandler, PreloadLinkHandler, LinkHandler } from './link-handler.service';
import { PrefetchRegistry } from './prefetch-registry.service';
import { QuicklinkStrategy } from './quicklink-strategy.service';
import * as i0 from "@angular/core";
export class QuicklinkModule {
}
QuicklinkModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.4", ngImport: i0, type: QuicklinkModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
QuicklinkModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.0.4", ngImport: i0, type: QuicklinkModule, declarations: [LinkDirective], exports: [LinkDirective] });
QuicklinkModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.0.4", ngImport: i0, type: QuicklinkModule, providers: [
        {
            provide: LinkHandler,
            useClass: ObservableLinkHandler,
            multi: true
        },
        {
            provide: LinkHandler,
            useClass: PreloadLinkHandler,
            multi: true
        },
        PrefetchRegistry,
        QuicklinkStrategy
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.4", ngImport: i0, type: QuicklinkModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LinkDirective],
                    providers: [
                        {
                            provide: LinkHandler,
                            useClass: ObservableLinkHandler,
                            multi: true
                        },
                        {
                            provide: LinkHandler,
                            useClass: PreloadLinkHandler,
                            multi: true
                        },
                        PrefetchRegistry,
                        QuicklinkStrategy
                    ],
                    exports: [LinkDirective]
                }]
        }] });
